File: java/Prog4test.java

import java.util.ArrayList;
import java.util.Scanner;
import java.io.*;
import java.util.Collections;

public class Prog4test {
    
    public static void main(String[] args) {

	String filename = args[0];
	int n = 0, e = 0, m = 0;
        float [][] prob = new float[1][1]; 

        try (Scanner fin = new Scanner(new File(filename))) {

		n = fin.nextInt();
		e = fin.nextInt();
		m = fin.nextInt();
	        prob = new float[n][m];
		for (int i = 0; i < n; i++)
		{
		    for (int j = 0; j < m; j++)
		      prob[i][j] = fin.nextFloat();
		}


	} catch (FileNotFoundException ex) {
		System.out.println("File not found");
	}

        ArrayList<Integer> result = new ArrayList<Integer>();
        float res = Assign.Assign(n, e, m, prob, result);
	System.out.print(filename + "  Base : Energy assignment : ");
        for (int i = 0; i < result.size(); i++) {
	 	System.out.print(result.get(i) + " ");
	}
	System.out.println("   Base prob : " + res);
		
        ArrayList<Integer> result2 = new ArrayList<Integer>();
        float res2 = Assign.Assign2(n, e, m, prob, result2);
	System.out.print(filename + "  Bonus : Energy assignment : ");
        for (int i = 0; i < result2.size(); i++) {
	 	System.out.print(result2.get(i) + " ");
	}
	System.out.println("   Bonus prob : " + res2);
     }

}

File: prog4test.cpp
#include <iostream>
#include <fstream>
#include <cmath>
#include <vector>

using namespace std;

pair< vector<int>, float  >  assign(int N, int E, int M, float **prob);
pair< vector<int>, float  >  assign2(int N, int E, int M, float **prob);

int main(int argc, char *argv[])
{
   ifstream ifile(argv[1]);

   int N, E, M;
   ifile >> N >> E >> M;
   float *prob[N];

   
   for (int i = 0; i < N ; i++)
	{
            prob[i] = new float[M];
	    for (int j= 0; j < M; j++)
		{
		    float s;
		    ifile >> s;
		    prob[i][j] = s;
		}
	}

   pair< vector<int>,  float> res = assign(N,E,M,prob); 
 
   cout << argv[1] << "   Base  : Energy assignment : ";
   for (int i = 0; i < res.first.size(); i++)
	{
 	    cout <<  res.first[i] << " ";
	}
   cout << "\n";
   cout << argv[1] << "   Base prob : " << res.second << endl;

   pair< vector<int>, float  >  res2 = assign2(N, E, M, prob);
 
   cout << argv[1] << "   Bonus : Energy assignment : ";
   for (int i = 0; i < res2.first.size(); i++)
	{
 	    cout <<  res2.first[i] << " ";
	}
   cout << "\n";
   cout << argv[1] << "   Bonus prob : " << res2.second << endl;

   return 0;
}
 
